provider/http.go

```diff
@@ -219,6 +219,41 @@ func (h *HTTPClient) signAndSubmit(tx string, skipPreFlight bool) (string, error
 	return response.Signature, nil
 }
 
+// signAndSubmitBatch signs the given transactions and submits them.
+func (h *HTTPClient) signAndSubmitBatch(transactions interface{}, opts SubmitOpts) (*pb.PostSubmitBatchResponse, error) {
+	if h.privateKey == nil {
+		return nil, ErrPrivateKeyNotFound
+	}
+	batchRequest := pb.PostSubmitBatchRequest{}
+	batchRequest.SubmitStrategy = opts.SubmitStrategy
+	for _, tx := range transactions.([]interface{}) {
+		oneRequest := pb.PostSubmitRequestEntry{}
+		oneRequest.SkipPreFlight = opts.SkipPreFlight
+		if txStr, ok := tx.(string); ok {
+			signedTxBase64, err := transaction.SignTxWithPrivateKey(txStr, *h.privateKey)
+			if err != nil {
+				return nil, err
+			}
+			oneRequest.Transaction = &pb.TransactionMessage{
+				Content: signedTxBase64,
+			}
+		} else if txMsg, ok := tx.(*pb.TransactionMessage); ok {
+			signedTxBase64, err := transaction.SignTxWithPrivateKey(txMsg.Content, *h.privateKey)
+			if err != nil {
+				return nil, err
+			}
+			oneRequest.Transaction = &pb.TransactionMessage{
+				Content:   signedTxBase64,
+				IsCleanup: txMsg.IsCleanup,
+			}
+		}
+
+		batchRequest.Entries = append(batchRequest.Entries, &oneRequest)
+	}
```

Comment:
Can you refactor this? Doesn't look clean to have this big chunk for each provider

Reply:
SignAndSubmit is like 15 lines with spaces, this is 30